llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -781,6 +782,15 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(PassBuilder &PB) {
           return onlyAllocateVGPRs;
         return nullptr;
       });
+
+  if (KernelInfoEndLTO) {
+    PB.registerFullLinkTimeOptimizationLastEPCallback(
+        [this](ModulePassManager &PM, OptimizationLevel Level) {
+          FunctionPassManager FPM;
+          FPM.addPass(KernelInfoPrinter(this));
```

Comment:
There's already a registerFullLinkTimeOptimizationLastEPCallback call above, move this into it

Reply:
Done.